src/poprox_storage/scrape_articles.py

```diff
@@ -0,0 +1,264 @@
+def scrape_others(url, section):
+    soup = request_and_parse(url)
+    articles = []
+
+    # Locate the main section of the page
+    main_section = soup.find("div", class_="PageListStandardH")
+    stories = main_section.find_all("div", class_="PageList-items-item")
+
+    # Scrape all the stories
+    for idx, story in enumerate(stories):
+        level = "Story 1" if idx == 0 else "Story 2"
+        articles.append(extract_article_info(story, section, level))
+
+    return articles
+
+
+# Function to scrape the Science section
+def scrape_science(url):
+    soup = request_and_parse(url)
+    articles = []
+
+    # Locate the main section of the page, with the left and right columns
+    main_section = soup.select_one("body > div.Page-content > main > div:nth-child(1)")
+    left_column = main_section.find("div", class_="PageListStandardH")
+    right_column = main_section.find("div", class_="PageListRightRailA")
+
+    for idx, story in enumerate(left_column.find_all("div", class_="PageList-items-item")):
+        level = "Story 1" if idx == 0 else "Story 2"
+        articles.append(extract_article_info(story, "Science", level))
+
+    for idx, story in enumerate(right_column.find_all("div", class_="PagePromo")):
+        level = "Story 2" if idx == 0 else "Story 3"
+        articles.append(extract_article_info(story, "Science", level))
+
+    return articles
+
+
+# URLs for different sections
+urls = {
+    "Front Page": "https://apnews.com/",
+    "US News": "https://apnews.com/us-news",
+    "Politics": "https://apnews.com/politics",
+    "Sports": "https://apnews.com/sports",
+    "Entertainment": "https://apnews.com/entertainment",
+    "Business": "https://apnews.com/business",
+    "Science": "https://apnews.com/science",
+    "Oddities": "https://apnews.com/oddities",
+}
+
+articles = []
+
+# Scrape the Front Page
+print("Scraping the Front Page...")
+front_page_articles = scrape_front_page("https://apnews.com/")
+articles.extend(front_page_articles)
+
+# Scrape all other sections but Science
+print("Scraping other sections...")
+for section, url in urls.items():
+    if section != "Front Page" and section != "Science":
+        section_articles = scrape_others(url, section)
+        articles.extend(section_articles)
+
+# Scrape the Science section
+print("Scraping the Science section...")
+science_articles = scrape_science("https://apnews.com/science")
+articles.extend(science_articles)
+
+# Create a DataFrame from the list of articles
+print(f"Scraping finished! Total articles scraped: {len(articles)}")
+print("Creating a DataFrame...")
+df = pd.DataFrame(articles)
+print(df.head())
+
+# Add a time stamp to the file name
+now = datetime.datetime.now()
+now = now.strftime("%Y-%m-%d_%H-%M-%S")
+
+# # Save to the folder "articles"
+# path_to_save = f'scraped-articles/apnews_articles_{now}.csv'
+# df.to_csv(path_to_save, index=False)
+# print(f'Articles saved to {path_to_save}')
+
+# Database Connection and Table Creation
+
```

Comment:
Everything from here down looks like it's inserting data in the database, but doing so very manually. You can write raw SQL like this and it will work okay, but it's hard to re-use and maintain. This library (`poprox-storage`) provides repository classes (like [this one for user clicks on articles](https://github.com/CCRI-POPROX/poprox-storage/blob/main/src/poprox_storage/repositories/clicks.py#L32)) that provide a plain Python interface for other parts of the POPROX code base to use when interacting with the database (and other storage like S3.)

Let's see if we can reshape this code in that direction. (More on that below.)